src/Promise.ts

```diff
@@ -473,8 +477,8 @@ export default class Promise<T> implements Thenable<T> {
 	/**
 	 * Adds a callback to the promise to be invoked when the asynchronous operation completes successfully.
 	 */
-	then<U>(onFulfilled?: (value?: T) => (U | Thenable<U>), onRejected?: (reason?: Error) => void): Promise<U>;
-	then<U>(onFulfilled?: (value?: T) => (U | Thenable<U>), onRejected?: (reason?: Error) => (U | Thenable<U>)): Promise<U> {
+	then<U>(onFulfilled?: ((value?: T) => (U | Thenable<U> | null | undefined)) | null | undefined, onRejected?: (reason?: Error) => void): Promise<U>;
```

Comment:
TypeScript [lib.es6.d.ts](https://github.com/Microsoft/TypeScript/blob/master/lib/lib.es6.d.ts#L5119) types these a bit differently to not need those.  Worth considering aligning to that?